devicetypes/fibargroup/fibaro-walli-double-switch.src/fibaro-walli-double-switch.groovy

```diff
@@ -0,0 +1,535 @@
+			commands += encap(zwave.configurationV2.configurationSet(scaledConfigurationValue: it.disableValue, parameterNumber: it.parameterNumber, size: it.size))
+			commands += encap(zwave.configurationV2.configurationGet(parameterNumber: it.parameterNumber))
+		}
+	}
+	sendHubCommand(commands)
+}
+
+private getPreferenceValue(preference, value = "default") {
+	def integerValue = value == "default" ? preference.defaultValue : value.intValue()
+	switch (preference.type) {
+		case "enum":
+			return String.valueOf(integerValue)
+		case "boolean":
+			return String.valueOf(preference.optionActive == integerValue)
+		default:
+			return integerValue
+	}
+}
+
+private getCommandValue(preference) {
+	def parameterKey = preference.key
+	switch (preference.type) {
+		case "boolean":
+			return settings."$parameterKey" ? preference.optionActive : preference.optionInactive
+		case "boolRange":
+			def parameterKeyBoolean = parameterKey + "Boolean"
+			return settings."$parameterKeyBoolean" ? settings."$parameterKey" : preference.disableValue
+		default:
+			return Integer.parseInt(settings."$parameterKey")
+	}
+}
+
+/**
+ * Mapping of command classes and associated versions used for this DTH
+ */
+private getCommandClassVersions() {
+	[
+		0x20: 1,  // Basic
+		0x25: 1,  // Switch Binary
+		0x30: 1,  // Sensor Binary
+		0x31: 2,  // Sensor MultiLevel
+		0x32: 3,  // Meter
+		0x56: 1,  // Crc16Encap
+		0x60: 3,  // Multi-Channel
+		0x70: 2,  // Configuration
+		0x72: 2,  // Manufacturer Specific
+		0x73: 1,  // Powerlevel
+		0x84: 1,  // WakeUp
+		0x86: 2,  // Version
+		0x98: 2   // Security
+	]
+}
+
+def configure() {
+	log.debug "Configure..."
+	response([
+			encap(zwave.multiChannelV3.multiChannelEndPointGet())
+	])
+}
+
+def parse(String description) {
+	def result = null
+	if (description.startsWith("Err")) {
+		result = createEvent(descriptionText:description, isStateChange:true)
+	} else if (description != "updated") {
+		def cmd = zwave.parse(description)
+		if (cmd) {
+			result = zwaveEvent(cmd, null)
+		}
+	}
+	log.debug "parsed '${description}' to ${result.inspect()}"
+	result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.configurationv2.ConfigurationReport cmd, enpoint = null) {
+	// Preferences template begin
+	log.debug "Configuration report: ${cmd}"
+	def preference = parameterMap.find( {it.parameterNumber == cmd.parameterNumber} )
+	def key = preference.key
+	def preferenceValue = getPreferenceValue(preference, cmd.scaledConfigurationValue)
+	if (settings."$key" == preferenceValue) {
+		state.currentPreferencesState."$key".value = settings."$key"
+		state.currentPreferencesState."$key".status = "synced"
+	} else if (preference.type == "boolRange") {
+		if (state.currentPreferencesState."$key".status == "disablePending" && preferenceValue == preference.disableValue) {
+			state.currentPreferencesState."$key".status = "disabled"
+		} else {
+			runIn(5, "syncConfiguration", [overwrite: true])
```

Comment:
Is `[overwrite: true]` necessary? @dkirker mentioned in another PR that was the default behavior.

Reply:
Even if that's the default behavior, I would leave [overwrite: true] - for clarity.